contracts/Tournament.sol

```diff
@@ -2,25 +2,44 @@
 pragma solidity ^0.8.10;
 
 contract Tournament {
+
+    enum State { Ongoing, BetsClosed, Failed, PaidOut }
+
     uint public immutable COMMISSION = 5;
     
     address public dealer;
     uint256 public buyIn;
+    uint256 public id;
+    State public state;
```

Comment:
What do we need that field for? It looks like you don't use it anywhere

Reply:
I thought that it would be a good idea to have some kind of id for the :
id -> address mapping on the backend side.

It was natural for me to assume that on the backend side we have to somehow distinguish which contracts were created by which users, hence it's good to distinguish them by using id / name fields.

It's purely for future development.

Reply:
How about introducing it in the future then? Let's follow YAGNI

Reply:
kay!

Reply:
Yes, u're right !